File: common/src/main/java/org/arnhold/plugins/MaDMPFileLoaderImpl.java
package org.arnhold.plugins;

import org.apache.jena.rdf.model.Model;
import org.example.dcsojson.DcsoJsonTransformer;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.stereotype.Component;
import org.springframework.stereotype.Service;

import java.io.File;

@Component
@Service
public class MaDMPFileLoaderImpl implements MaDMPLoader {

    @Autowired
    DcsoJsonTransformer dcsoJsonTransformer;

    @Override
    public Model fromIdentifier(File identifier) {
        try {
            return dcsoJsonTransformer.convertPlainToModel(identifier);
        } catch (Exception e) {
            // TODO replace with proper error handling
            throw new RuntimeException(e);
        }
    }
}

File: case-study/src/main/kotlin/org/arnhold/dmpeval/casestudy/loader/MaDMPFileLoaderImpl.kt
package org.arnhold.dmpeval.casestudy.loader

import at.ac.tuwien.dcsojson.DcsoJsonTransformer
import org.apache.jena.rdf.model.Model
import org.arnhold.sdk.dmpLoader.MaDMPLoader
import org.springframework.beans.factory.annotation.Autowired
import org.springframework.stereotype.Component
import org.springframework.stereotype.Service
import java.io.File

@Component
@Service
class MaDMPFileLoaderImpl @Autowired constructor(var dcsoJsonTransformer: DcsoJsonTransformer) : MaDMPLoader {

    override fun fromIdentifier(identifier: File): Model {
        try {
            return dcsoJsonTransformer.convertPlainToModel(identifier)
        } catch (e: Exception) {
            throw RuntimeException(e)
        }
    }
}
